src/main/java/de/cronn/reflection/util/RecordSupport.java

```diff
@@ -0,0 +1,289 @@
+package de.cronn.reflection.util;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.jetbrains.annotations.Nullable;
+import org.objenesis.ObjenesisHelper;
+
+import net.bytebuddy.ByteBuddy;
+
+class RecordSupport {
+
+	private static volatile Boolean hasIsRecordMethod = null;
+
+	private static final ClassValue<Class<?>> dummySubclasses = new ClassValue<Class<?>>() {
+		@Override
+		protected Class<?> computeValue(Class<?> type) {
+			return new ByteBuddy()
+				.subclass(type)
+				.make()
+				.load(RecordSupport.class.getClassLoader())
+				.getLoaded();
+		}
+	};
+
+	static boolean isRecord(Class<?> beanClass) {
+		if (hasIsRecordMethod == null) {
```

Comment:
I'm just curious why not to use `Class.forName("java.lang.Record")`, 

so usage can be (simplified) 
```java
if(hasRecordClass==null){
  try{
     this.recordClass = Class.forName("java.lang.Record");
     this.hasRecordClass = true;
     
  } catch (ClassNotFoundException e){
    this.hasRecordClass = false;
  }
}

if(hasRecordClass){
  return this.recordClass.isAssignableFrom(beanClass);
}
return false;
```

Less reflection usage. Maybe better performance, but I guess current approach is also not costly.

Reply:
Thanks for your review and this suggestion!

I actually ran a quick benchmark to compare
```java
Class<?> recordClass = Class.forName("java.lang.Record");
recordClass.isAssignableFrom(beanClass);
```
vs.
```java
Method isRecordMethod = Class.class.getMethod("isRecord");
isRecordMethod.invoke(beanClass);
```

On a **cold** JVM (Java 17.0.1) the first approach is about an order of magnitude faster on my machine:

- ~50 µs for `Class.forName` + ~5 µs for the `isAssignableFrom` call
vs.
- ~550 µs for `Class.class.getMethod` + ~90 µs for `isRecordMethod.invoke`

As you already mentioned, things are _much_ faster once the JVM is warmed up and the JIT compiler kicked in and things got cached. On my machine the numbers go even down to an impressive ~1.2 µs  ("Class.forName") vs. ~3 µs ("getMethod").

I will apply your suggestion to use `Class.forName` since the code complexity is about the same but it’s always faster. Thanks!

BTW: I miss your good code reviews at work ;)

Reply:
I miss your reviews too :)